packages/core/admin/admin/src/content-manager/services/relations.ts

```diff
@@ -111,8 +113,10 @@ const relationsApi = contentManagerApi.injectEndpoints({
            * Relations will always have unique IDs, so we can therefore assume
            * that we only need to push the new items to the cache.
            */
-          const existingIds = currentCache.results.map((item) => item.id);
-          const uniqueNewItems = newItems.results.filter((item) => !existingIds.includes(item.id));
+          const existingIds = currentCache.results.map((item) => item.documentId);
+          const uniqueNewItems = newItems.results.filter(
+            (item) => !existingIds.includes(item.documentId)
+          );
```

Comment:
relations will not have unique ids if we use a documentId (there are scenarios when we show multiple locales of the same documentID)

wonder if in cases like this we want to also return the entry id

Reply:
When do we show multiple locales? Do you mean if we query from a non i18n content type to a target i18n content type?

I don't think the endpoint is built to handle that situation at all atm e.g.

https://github.com/strapi/strapi/blob/a5be926572b9790b414c43810a30d494a4d08e0f/packages/core/content-manager/server/src/controllers/relations.ts#L277-L293

Reply:
exactly that scenario!

Also, Why not always fetch draft and use the documentMetadata to get the status of those relations/

Reply:
Done in [`37eb0fa` (#19681)](https://github.com/strapi/strapi/pull/19681/commits/37eb0faf2e66c355aae1efec83f353c76907e750)